pkg/sql/distsql_physical_planner_test.go

```diff
@@ -1528,26 +1625,42 @@ func TestPartitionSpans(t *testing.T) {
 					tc.partitionState, *planCtx.spanPartitionState)
 			}
 
-			resMap := make(map[int][][2]string)
-			for _, p := range partitions {
-				if _, ok := resMap[int(p.SQLInstanceID)]; ok {
-					t.Fatalf("node %d shows up in multiple partitions", p.SQLInstanceID)
-				}
-				var spans [][2]string
-				for _, s := range p.Spans {
-					spans = append(spans, [2]string{string(s.Key), string(s.EndKey)})
-				}
-				resMap[int(p.SQLInstanceID)] = spans
-			}
-
 			recording := getRecAndFinish()
 			for _, expectedMsg := range tc.partitionStates {
 				require.NotEqual(t, -1, tracing.FindMsgInRecording(recording, expectedMsg))
 			}
 
+			resMap := stringifyPartition(partitions)
+
 			if !reflect.DeepEqual(resMap, tc.partitions) {
 				t.Errorf("expected partitions:\n  %v\ngot:\n  %v", tc.partitions, resMap)
 			}
+
+			// Test strict locality filtering if applicable.
+			if len(locFilters) > 0 {
+				ctx := context.Background()
```

Comment:
don't you already have a ctx and evalCtx in scope? why shadow with identical redeclarations?

Reply:
i needed to redeclare them to avoid some spanUseAfter finish panics. I've renamed them and added comment to clarify this.